src/modules/Auth/api.tsx

```diff
@@ -0,0 +1,71 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import { LoginCredentials, User, UserGet, UserResponse } from './types';
+import { SignUpRequest, UserSignUpResponse } from './types';
+import storage from '../../utils/storage';
+
+// Define a service using a base URL and expected endpoints
+export const authApi = createApi({
+  reducerPath: 'authApi',
+  baseQuery: fetchBaseQuery({ baseUrl: 'http://localhost:3001/api/v1/',
+  prepareHeaders: (headers, { getState }) => {
+    const token = storage.getToken();
```

Comment:
Token should be accessed from localstorage only once. Accessing it again & again is a expensive operation